olive/cache.py

```diff
@@ -113,3 +114,44 @@ def clean_pass_run_cache(pass_type: str, cache_dir: Union[str, Path] = ".olive-c
     run_jsons = list(run_cache_dir.glob(f"{pass_type}-*.json"))
     for run_json in run_jsons:
         _delete_run(run_json.stem, cache_dir)
+
+
+def save_model(
```

Comment:
Since we are saving a model from cache into a user specified location, I think it is still something like `save_model`? It's already under `olive.cache` so I don't think there is too much ambiguity about its use?

Reply:
Should I rename it to `dump_model`?

Reply:
or `persist_model`?

Reply:
Not the blocker for this PR. Feel free to use the one you like. Just leave some comments for better understanding.